workflows/Utility/modules/LogCombiner/workflow/Snakefile

```diff
@@ -0,0 +1,65 @@
+"""LogCombiner-BEAST1
+
+This module takes two log or tree files from BEAST (v1.10.4/5) as input and returns a single combined log or tree file as output. Please see https://beast.community/logcombiner for more information.
+
+Params:
+    trees (boolean): use this option to combine tree log files (default is false)
+    decimal (boolean): this option converts numbers from scientific to decimal notation (default is false)
+    burnin (integer): number of states to be considered as 'burn-in' (default is 100000)
+    resample (integer): resample the log files to this frequency (the original sampling frequency must be a factor of this value) (default is 10000)
+    scale (float): a scaling factor that will multiply any time units by this value
+    strip (boolean): strip out all annotations (trees only)
+    renumber (boolean): this option renumbers output states consecutively (default is false)
+    input:
+        infile1 (string): input log/tree file 1
+        infile2 (string): input log/tree file 2
+    output:
+        outfile (string): output combined log/tree file
+"""
+
+configfile: "config/config.yaml"
+
+indir=config["input_namespace"]
+outdir=config["output_namespace"]
+params=config["params"]
+
+rule target:
+    input:
+        infile1 = expand(
+            "results/{indir}/{filename}",
+            indir=indir,
+            filename=params["input"]["infile1"]
+        ),
+        infile2 = expand(
+            "results/{indir}/{filename}",
+            indir=indir,
+            filename=params["input"]["infile2"]
+        )
+    output:
+        outfile = expand(
+            "results/{outdir}/{filename}",
+            outdir=outdir,
+            filename=params["output"]["outfile"]
+        )
+    conda:
+        "envs/conda.yaml"
+    params:
+        trees=params["trees"],
+        decimal=params["decimal"],
+        burnin=params["burnin"],
+        resample=params["resample"],
+        scale=params["scale"],
+        strip=params["strip"],
+        renumber=params["renumber"] 
+    shell:
+        """
+        logcombiner \
+            $(if [ {params.trees} == true ]; then echo "-trees"; else echo ""; fi) \
```

Comment:
I moved this logic to the `params` section in `TreeAnnotator` as it seems simpler than dealing with nested '{'-'}' issues (since '{}' can be variables, wildcards, or regular bash braces in the `shell` directive). Either works, but useful to be aware of moving forwards.

Reply:
That makes sense, thanks